packages/flutter_tools/test/general.shard/ios/metal_api_validator_test.dart

```diff
@@ -0,0 +1,106 @@
+
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/ios/migrations/metal_api_validation_migration.dart';
+import 'package:flutter_tools/src/project.dart';
+import 'package:test/fake.dart';
+
+import '../../src/common.dart';
+
+
+void main() {
+  testWithoutContext('Adds Metal API setting to matching file', () {
+    final FileSystem fs = MemoryFileSystem.test();
+
+    final File file = fs.file('test_file')
+      ..createSync()
+      ..writeAsStringSync('''
+<?xml version="1.0" encoding="UTF-8"?>
+  <LaunchAction
+    buildConfiguration = "Debug"
+    selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
+    selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
+    launchStyle = "0"
+    useCustomWorkingDirectory = "NO"
+    ignoresPersistentStateOnLaunch = "NO"
+    debugDocumentVersioning = "YES"
+    debugServiceExtension = "internal"
+    allowLocationSimulation = "YES">
+''');
+    final FakeIosProject project = FakeIosProject(file);
+    final MetalAPIValidationMigrator validator = MetalAPIValidationMigrator(project, BufferLogger.test());
+
+    expect(() async => validator.migrate(), returnsNormally);
+
+    expect(file.readAsStringSync(), contains(
+      'debugServiceExtension = "internal"'
+    '\n    enableGPUValidationMode = "1"'));
```

Comment:
Can you check the entire XML file to make sure other whitespace doesn't get messed up.
```
'''
<?xml version="1.0" encoding="UTF-8"?>
  <LaunchAction
    buildConfiguration = "Debug"
    selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
    selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
    launchStyle = "0"
    useCustomWorkingDirectory = "NO"
    ignoresPersistentStateOnLaunch = "NO"
    debugDocumentVersioning = "YES"
    debugServiceExtension = "internal"
    enableGPUValidationMode = "1"
    allowLocationSimulation = "YES">
'''
```

Reply:
oh the white space is definitely messed up, since the indentation matters. I'm wagering that most folks never actually open this file...

Reply:
Ideally given the normal generated template scheme that's never otherwise been updated, the migration adds the right whitespace.  Otherwise Xcode will try to format it if the whitespace is off the first time it opens, which will dirty the working copy again.

Reply:
I'm a real programmer and regex'd my way to a solution.